dpppt-backend-sdk/dpppt-backend-sdk-interops/src/main/java/org/dpppt/backend/sdk/interops/insertmanager/insertionfilters/EfgsDsosFilter.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2020 Ubique Innovation AG <https://www.ubique.ch>
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package org.dpppt.backend.sdk.interops.insertmanager.insertionfilters;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import org.dpppt.backend.sdk.model.gaen.GaenKeyForInterops;
+import org.dpppt.backend.sdk.utils.UTCInstant;
+
+/**
+ * This filter allows to drop/ignore certain keys based on DSOS (Days Since Onset of Symptoms)
+ * values when downloading keys from the EFGS hub. The filter itself can be enabled/disabled in the
+ * configuration of DP3T. Also the thresholds itself can be defined in the configuration.
+ */
+public class EfgsDsosFilter implements InteropsKeyInsertionFilter {
+  private static int DSOS_RANGE_MAX = 19;
+
+  int symptomaticOnsetKnownDropDaysBeforeOnset;
+  int symptomaticOnsetRangeDropDaysBeforeRangeStart;
+  int symptomaticUnknownOnsetDropDaysBeforeSubmission;
+  int asymptomaticDropDaysBeforeSubmission;
+  int unknownSymptomStatusDropDaysBeforeSubmission;
+
+  public EfgsDsosFilter(
+      int symptomaticOnsetKnownDropDaysBeforeOnset,
+      int symptomaticOnsetRangeDropDaysBeforeRangeStart,
+      int symptomaticUnknownOnsetDropDaysBeforeSubmission,
+      int asymptomaticDropDaysBeforeSubmission,
+      int unknownSymptomStatusDropDaysBeforeSubmission) {
+
+    this.symptomaticOnsetKnownDropDaysBeforeOnset = symptomaticOnsetKnownDropDaysBeforeOnset;
+    this.symptomaticOnsetRangeDropDaysBeforeRangeStart =
+        symptomaticOnsetRangeDropDaysBeforeRangeStart;
+    this.symptomaticUnknownOnsetDropDaysBeforeSubmission =
+        symptomaticUnknownOnsetDropDaysBeforeSubmission;
+    this.asymptomaticDropDaysBeforeSubmission = asymptomaticDropDaysBeforeSubmission;
+    this.unknownSymptomStatusDropDaysBeforeSubmission =
+        unknownSymptomStatusDropDaysBeforeSubmission;
+  }
+
+  @Override
+  public List<GaenKeyForInterops> filter(UTCInstant now, List<GaenKeyForInterops> content) {
+    return content.stream().filter(key -> !dropKeyDueToDsosValue(key)).collect(Collectors.toList());
```

Comment:
would it be easier to understand, if we change this to something like "keyIsWithValidDsos" in order to drop the "!" ?